test/mailers/patient_mailer_test.rb

```diff
@@ -612,4 +612,31 @@ def twilio_error
       assert patient.updated_at < 290.days.ago
     end
   end
+
+  test 'assessment_email creates an assessment_email_error history when it fails' do
+    ActionMailer::Base.deliveries.clear
+    patient = create(:patient,
+                     preferred_contact_method: 'E-mailed Web Link',
+                     email: 'testpatient@example.com')
+    original_updated_at = patient.updated_at
+    allow_any_instance_of(Patient).to(receive(:select_language).and_raise('Testing assessment_email'))
+    assert_difference 'patient.histories.length', 1 do
+      PatientMailer.assessment_email(patient).deliver_now
+      patient.reload
+      assert_equal('Assessment Email Error', patient.histories.first.history_type)
+      assert_equal(patient.updated_at, original_updated_at)
+      assert_equal(ActionMailer::Base.deliveries.length, 0)
+    end
+  end
+
+  test 'assessment_email logs to sentry when it fails' do
+    ActionMailer::Base.deliveries.clear
+    patient = create(:patient,
+                     preferred_contact_method: 'E-mailed Web Link',
+                     email: 'testpatient@example.com')
+    allow(Raven).to receive(:capture_exception)
+    PatientMailer.assessment_email(patient).deliver_now
```

Comment:
In the other added test in this file it looks like you're using `allow_any_instance_of(Patient).to(receive(:select_language).and_raise('Testing assessment_email'))` to trigger an exception and subsequent history item. I'm not totally following why it isn't necessary here as well since the Sentry capture is in the same catch block as the new history item.

Reply:
I looked deeper into this and it looks like it's hitting the exception because the patient submission token is nil. It's actually possible to remove L622 and have that test still pass because of this reason. 

I think both of these new tests should include an argument for the submission token and this test should also include the line `allow_any_instance_of(Patient).to(receive(:select_language).and_raise('Testing assessment_email'))`. This makes it much easier to understand how the test is triggering an exception.

```ruby
  test 'assessment_email creates an assessment_email_error history when it fails' do
    ActionMailer::Base.deliveries.clear
    patient = create(:patient,
                     preferred_contact_method: 'E-mailed Web Link',
                     email: 'testpatient@example.com',
                     submission_token: SecureRandom.urlsafe_base64[0, 10])
    original_updated_at = patient.updated_at
    allow_any_instance_of(Patient).to(receive(:select_language).and_raise('Testing assessment_email'))
    assert_difference 'patient.histories.length', 1 do
      PatientMailer.assessment_email(patient).deliver_now
      patient.reload
      assert_equal('Assessment Email Error', patient.histories.first.history_type)
      assert_equal(patient.updated_at, original_updated_at)
      assert_equal(ActionMailer::Base.deliveries.length, 0)
    end
  end

  test 'assessment_email logs to sentry when it fails' do
    ActionMailer::Base.deliveries.clear
    patient = create(:patient,
                     preferred_contact_method: 'E-mailed Web Link',
                     email: 'testpatient@example.com',
                     submission_token: SecureRandom.urlsafe_base64[0, 10])
    allow_any_instance_of(Patient).to(receive(:select_language).and_raise('Testing assessment_email'))
    allow(Raven).to receive(:capture_exception)
    PatientMailer.assessment_email(patient).deliver_now
    expect(Raven).to have_received(:capture_exception)
    assert_equal(ActionMailer::Base.deliveries.length, 0)
  end
```

Reply:
https://github.com/SaraAlert/SaraAlert/pull/990/commits/7c2b1f3785790d93068b323ec9d6cd8a57dc0869